### PR TITLE
Add workaround when dependency errs or is not found

### PR DESCRIPTION
**Pull request by nickiemc**
Makes updates to initially resolve the following issues:

- #27 (https://github.com/chriskiehl/Gooey)
- #32 (https://github.com/HelloZeroNet/ZeroNet)
- #33 (https://github.com/diaspora/diaspora)
- #35 (https://github.com/gitlabhq/gitlabhq)
- #36 (https://github.com/discourse/discourse)

Note - these updates simply allow Freshli to successfully complete a run against the given repositories. The current logic simply skips problematic dependencies. These problematic dependencies still need to be investigated separately.

### CODE REVIEW

File: Freshli/LibYearCalculator.cs [hunk 1]
```diff
@@ -36,9 +43,9 @@ IManifest manifest
           _logger.Trace(
             $"Package({package.Name}, {package.Version}): " +
             $"current = {currentVersion.ToSemVer()}" +
-            $"@{currentVersion.DatePublished}, " +
+            $"@{currentVersion.DatePublished:d}, " +
             $"latest = {latestVersion.ToSemVer()}" +
-            $"@{latestVersion.DatePublished}"
+            $"@{latestVersion.DatePublished:d}"
```
Comment (nickiemc) [inline Freshli/LibYearCalculator.cs:48]: Reduces noise by updating date format from 

```
current = 6.0.@5/18/2020 12:00:00 AM, latest = 6.0.@6/17/2020 12:00:00 AM
```

to 
```
current = 6.0.@5/18/2020, latest = 6.0.@6/17/2020
```

File: Freshli/LibYearCalculator.cs [hunk 2]
```diff
@@ -19,7 +19,14 @@ IManifest manifest
...
       foreach (var package in Manifest) {
-        var latestVersion = Repository.LatestAsOf(date, package.Name);
+        VersionInfo latestVersion;
+        try {
+          latestVersion = Repository.LatestAsOf(date, package.Name);
+        } catch (Exception e) {
+          _logger.Info($"Skipping {package.Name}: {e.Message}");
+          continue;
+        }
+
```
Comment (nickiemc) [inline Freshli/LibYearCalculator.cs:29]: I'm currently handling all exceptions the same, but that may change, hence throwing two types of exceptions.

**Comment by mscottford** [comment]
I think this is looking good so far! Thanks so much.

**Comment by nickiemc** [comment]
@mscottford Based on our on conversation today, I'm going to go ahead and merge this as is.

**Pull request merged**